src/main/java/com/ably/kafka/connect/client/DefaultAblyClient.java

```diff
@@ -59,6 +67,21 @@ public void publishFrom(SinkRecord record) throws ConnectException {
             //this exception should cause the calling task to abort
             throw new ConnectException("Cannot publish to Ably when connection failed");
         }
+        //check both channel and message config for skipping the record
+        final String messageConfig = connectorConfig.getString(MESSAGE_CONFIG);
```

Comment:
Too wide variable scope: As far as I can see `messageConfig` and `channelConfig` are only used if `skipOnKeyAbsence` is `true`, so I would move these two `connectorConfig.getString()` calls to within the check below.

Reply:
Good point - I also moved this to a new method (for now) as this method was getting bigger. Change : 02939fd5e7f6ec18954639a8d1a6e687c3393212